### DISCUSSION TITLE
[Feature] Update Album Thumbnails to Latest

### DISCUSSION THREAD

**Discussion by cfitzw** [immich-app/immich#3493] [Feature Request]
### The feature

After adding photos to album, the thumbnail for that album should be set/updated to be the most recent photo, except when a custom thumbnail  has already been set for that album.

From what I can tell, when setting a custom album "cover"/thumbnail, nothing gets stored in the database to denote that it was manually set (and therefore no way to code against it). 

Also, upon upload of files for a new album, it looks to take the first photo in the array versus do some comparisons to determine and set to the latest.

My thought is that a new column should be created in the albums db table named 'isAlbumThumbnailOverridden'...

- Upon a user setting a custom thumbnail, the boolean gets updated to true and the asset id stored alongside it (the latter already exists). 
- If boolean is set to false (default), then every time new photos get added to an album, the thumbnail gets updated to be the most recent pic (between existing album photos and new array).
- If a album has a custom set thumbnail, but that asset it removed from the album, update the thumbnail to be the most recent pic from the album.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by Dusky-dev** [comment]
+1
